ansible/roles/telegraf/tasks/main.yml

```diff
@@ -0,0 +1,80 @@
+---
+# Copyright (C) 2021 Dr. Serge Victor <https://dr.sergevictor.eu/>
+# Copyright (C) 2020 DebOps <https://debops.org/>
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+- name: Import custom Ansible plugins
+  import_role:
+    name: 'ansible_plugins'
+
+- name: Import DebOps global handlers
+  import_role:
+    name: 'global_handlers'
+
+- name: Divert the original telegraf main configuration file
+  dpkg_divert:
+    path: '/etc/telegraf/telegraf.conf'
+    state: 'present'
+    delete: true
+  notify: [ 'Restart Telegraf' ]
+
+- name: Configure Telegraf instance leading file
+  template:
+    src: 'etc/telegraf/telegraf.conf.j2'
+    dest: '/etc/telegraf/telegraf.conf'
+    owner: 'root'
+    group: 'root'
+    mode: '0644'
+  notify: [ 'Restart Telegraf' ]
+
+- name: Remove plugins configuration if requested
+  file:
+    path: '/etc/telegraf/telegraf.d/{{ item.name }}.conf'
+    state: 'absent'
+  loop: '{{ telegraf__configuration_combined | parse_kv_items }}'
+  when: item.state|d('present') == 'absent' and item.config|d()
+  notify: [ 'Restart Telegraf' ]
+
+- name: Find plugins which are in filesystem
+  find:
+    paths: /etc/telegraf/telegraf.d
+    patterns: '*.conf'
+  register: telegraf_existing_plugins_filenames
```

Comment:
You should use namespacing and mark the variables as registers properly, so that they are eaiser to locate in the code. Rename this variable to `telegraf__register_existing_plugins_filenames`.